### DISCUSSION TITLE
Request to make data pointer const in C++ new_from_memory_copy()

### DISCUSSION THREAD

**Discussion by VivitionDeveloper** [libvips/libvips#4124] [General]
It may be nice to make the C++ [new_from_memory_copy](https://www.libvips.org/API/current/cpp/classVImage.html#a4528866ca838d28c00e8220c9ef3bb89) data pointer const, as the underlying code calls [vips_image_new_from_memory_copy](https://www.libvips.org/API/current/VipsImage.html#vips-image-new-from-memory-copy), where the data pointer is already const, see . It then clearly expresses that the copy keeps its fingers off from the provided data pointer. Just as an suggestion, as I know have to perform a const_cast in our own code.

With this requested change, it also makes clear that new_from_memory (without copy) uses the provided data buffer (with responsibility to the caller for memory management).

I'm not sure whether I posted this question in the right category.

**Comment by kleisauke** [comment]
This was already done with commit https://github.com/libvips/libvips/commit/75174cd811ee2256b065c6f6dcfd18de4877eff9.

**Reply by VivitionDeveloper** [reply]
Thank you.
